### PR TITLE
Expand anndata compatibility

### LINKED ISSUE #114: Allow wider anndata version
`pydeseq2` currently pins anndata to 0.8.0. anndata 0.9.1 is now out, would you consider testing it with the newer version?

### PR DESCRIPTION
**Pull request by scottgigante-immunai**
#### Reference Issue or PRs
Fixes #114 

#### What does your PR implement? Be specific.
Expands anndata compatibility

### CODE REVIEW

**Comment by BorisMuzellec** [comment]
Hi @scottgigante-immunai, thanks for opening this PR!

I agree that it would be best to use the latest version of `anndata`.

Unfortunately I probably won't have much time to work on this in the next few days/weeks, but if you feel like investigating why tests fail, I'd be happy to review this PR once it passes tests. In the meantime, I can also help to some extent.

**Comment by scottgigante-immunai** [comment]
Thanks @BorisMuzellec ! Looks like I need your approval to run the tests, but otherwise I'm happy to debug.

**Comment by scottgigante-immunai** [comment]
@BorisMuzellec this should be good to go :)

**Review by BorisMuzellec — CHANGES_REQUESTED** [review]

File: pydeseq2/dds.py [hunk 1]
```diff
@@ -175,21 +175,20 @@ def __init__(
         joblib_verbosity: int = 0,
     ) -> None:
 
-        if adata is None and (counts is None or clinical is None):
-            raise ValueError(
-                "Either adata or both counts and clinical arguments must be provided."
-            )
-
```
[the remark this answers is not in the document]
Comment (scottgigante-immunai) [reply pydeseq2/dds.py:182]: I didn't remove this, I just moved it down -- as written, the current code doesn't guard against _both_ being set, only against _neither_ being set. I'm happy to put that in! I just moved the _neither_ part down to keep mypy happy.

[the remark this answers is not in the document]
Comment (BorisMuzellec) [reply pydeseq2/dds.py:182]: Indeed, for some reason I assumed there was a test to guard agains both being set :sweat_smile:. 

Thanks a lot for implementing this!

**Pull request merged**